### PR TITLE
[refactor] Updated GraphQL use of UrlResolver to Route

### PR DESCRIPTION
**Pull request by jcalcaben**
## Description

Updates the field Query.urlResolver, which is deprecated, to use the route query instead.

## Related Issue

Closes PWA-2324

## Acceptance

Any developer

### Verification Stakeholders

Any developer

## Verification Steps

1. Run dev server: `yarn watch:venia`
2. Verify application runs with no logging errors
3. Click through home page, category page, and product page
4. Verify no logging errors
5. Use store switcher
6. Verify store switching still works as expected

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/peregrine/lib/talons/Header/storeSwitcher.gql.js [hunk 1]
```diff
@@ -11,11 +11,20 @@ export const GET_STORE_CONFIG_DATA = gql`
...
-export const GET_URL_RESOLVER_DATA = gql`
-    query getUrlResolverData($url: String!) {
-        urlResolver(url: $url) {
-            id
+export const GET_ROUTE_DATA = gql`
+    query getRouteData($url: String!) {
+        route(url: $url) {
             type
+            ... on CmsPage {
+                identifier
```
Comment (fooman) [inline packages/peregrine/lib/talons/Header/storeSwitcher.gql.js:19]: Might want to pick up the changes coming from here
https://github.com/magento/pwa-studio/pull/3580/files

File: packages/peregrine/lib/talons/Header/storeSwitcher.gql.js [hunk 2]
```diff
@@ -11,11 +11,20 @@ export const GET_STORE_CONFIG_DATA = gql`
...
+            }
+            ... on ProductInterface {
+                id
```
Comment (pedrochiossi) [inline packages/peregrine/lib/talons/Header/storeSwitcher.gql.js:22]: as @fooman pointed out, these should be uid

File: packages/peregrine/lib/talons/Header/storeSwitcher.gql.js [hunk 3]
```diff
@@ -11,11 +11,20 @@ export const GET_STORE_CONFIG_DATA = gql`
...
+                __typename
+            }
+            ... on CategoryInterface {
+                id
```
Comment (pedrochiossi) [inline packages/peregrine/lib/talons/Header/storeSwitcher.gql.js:26]: uid

**Review by pedrochiossi — APPROVED** [review]

**Comment by mikhaelbois** [comment]
@jcalcaben 
Please disable linting rules in `packages/peregrine/lib/talons/Header/storeSwitcher.gql.js` and update with `develop`.
`# eslint-disable-next-line @graphql-eslint/require-id-when-available`

Root components still work as needed.
However, now that we send the `uid` to the Category Root Component, we should update the code to reflect that change.
We should also update how we get the Category data to use the `category_uid` filter instead of `ids`.
Like you've suggested, we should open a follow-up task to do those updates.

**Comment by mikhaelbois** [comment]
QA ✅

**Review by jeremycharron — APPROVED** [review]

**Pull request merged**